src/component/mxgraph/MxGraphRenderer.ts

```diff
@@ -15,44 +15,41 @@
  */
 import Shape from '../../model/bpmn/internal/shape/Shape';
 import Edge from '../../model/bpmn/internal/edge/Edge';
-import BpmnModel from '../../model/bpmn/internal/BpmnModel';
-import ShapeBpmnElement, { ShapeBpmnSubProcess } from '../../model/bpmn/internal/shape/ShapeBpmnElement';
+import ShapeBpmnElement from '../../model/bpmn/internal/shape/ShapeBpmnElement';
 import Waypoint from '../../model/bpmn/internal/edge/Waypoint';
 import Bounds from '../../model/bpmn/internal/Bounds';
 import ShapeUtil from '../../model/bpmn/internal/shape/ShapeUtil';
 import CoordinatesTranslator from './renderer/CoordinatesTranslator';
 import StyleConfigurator from './config/StyleConfigurator';
 import { MessageFlow } from '../../model/bpmn/internal/edge/Flow';
 import { MessageVisibleKind } from '../../model/bpmn/internal/edge/MessageVisibleKind';
-import { ShapeBpmnMarkerKind } from '../../model/bpmn/internal/shape';
 import { BpmnMxGraph } from './BpmnMxGraph';
 import { LoadOptions } from '../options';
+import { RenderedModel } from '../registry/bpmn-model-registry'; // for types
 import { mxgraph } from './initializer';
 import { mxCell } from 'mxgraph'; // for types
 
 export default class MxGraphRenderer {
   constructor(readonly graph: BpmnMxGraph, readonly coordinatesTranslator: CoordinatesTranslator, readonly styleConfigurator: StyleConfigurator) {}
 
-  public render(bpmnModel: BpmnModel, loadOptions?: LoadOptions): void {
-    this.insertShapesAndEdges(bpmnModel);
+  public render(renderedModel: RenderedModel, loadOptions?: LoadOptions): void {
+    this.insertShapesAndEdges(renderedModel);
     this.graph.customFit(loadOptions?.fit);
   }
 
-  private insertShapesAndEdges(bpmnModel: BpmnModel): void {
-    const displayedModel = toDisplayedModel(bpmnModel);
-
+  private insertShapesAndEdges(renderedModel: RenderedModel): void {
```

Comment:
ℹ️  To avoid to have `renderedModel.pools` in the method, you can do something like this:
```suggestion
  private insertShapesAndEdges({pools, lanes, subprocesses, otherFlowNodes, boundaryEvents, edges}: RenderedModel): void {
```